components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt

```diff
@@ -292,10 +292,36 @@ class GeckoEngine(
      * See [Engine.listInstalledWebExtensions].
      */
     override fun listInstalledWebExtensions(onSuccess: (List<WebExtension>) -> Unit, onError: (Throwable) -> Unit) {
-        // TODO https://github.com/mozilla-mobile/android-components/issues/4500
+        // TODO https://bugzilla.mozilla.org/show_bug.cgi?id=1600742
         onSuccess(emptyList())
     }
 
+    /**
+     * See [Engine.enableWebExtension].
+     */
+    override fun enableWebExtension(
+        extension: WebExtension,
+        onSuccess: (WebExtension) -> Unit,
+        onError: (Throwable) -> Unit
+    ) {
+        // TODO https://bugzilla.mozilla.org/show_bug.cgi?id=1599585
+        webExtensionDelegate?.onEnabled(extension)
+        onSuccess(extension)
```

Comment:
I guess it is TODO but I find it odd that we pass `onSuccess` even though `webExtensionDelegate` is nullable.

Reply:
Yeaj, so, I think the `delegate` will always remain nullable, well eventually we will get rid of it, once we have the store available in the engine. The only reason I am calling `onSuccess` now is to be able test integration.